product_multi_barcode/models/product_product.py

```diff
@@ -0,0 +1,65 @@
+# © 2012-2014 Guewen Baconnier (Camptocamp SA)
+# © 2015 Roberto Lizana (Trey)
+# © 2016 Pedro M. Baeza
+# © 2018 Xavier Jimenez (QubiQ)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    barcode_ids = fields.One2many(
+        comodel_name="product.barcode",
+        inverse_name="product_id",
+        string="Barcodes",
+    )
+    barcode = fields.Char(
+        string="Main barcode",
+        compute="_compute_barcode",
+        store=True,
+        inverse="_inverse_barcode",
+        compute_sudo=True,
+    )
+
+    @api.depends("barcode_ids.name", "barcode_ids.sequence")
+    def _compute_barcode(self):
+        for product in self:
+            product.barcode = product.barcode_ids[:1].name
+
+    def _inverse_barcode(self):
+        for product in self:
+            if product.barcode_ids:
```

Comment:
@vib-adhoc , maybe the comparison must be more specific including [:1]. The same for the elif...

Reply:
That's not going to make any difference I think.